python/tank/util/shotgun/publish_resolve.py

```diff
@@ -79,41 +79,88 @@ def resolve_publish_path(tk, sg_publish_data):
+    - If you wanted to use more than one environment variable, this is possible
+      by extending the above syntax with a name, for example ``SHOTGUN_PATH_LINUX_SECONDARY``.
 
-    - Next, any environment variables of the form ``SHOTGUN_PATH_WINDOWS|MAC|LINUX``
-      or ``SHOTGUN_PATH_WINDOWS|MAC|LINUX_STORAGENAME`` will be added
-      to the translation table.
+    - In addition, all existing local storage definitions will be downloaded
+      and used in the same way.
 
-    - The ``file://`` path will be compared against all the existing roots
-      for all operating systems. The first match detected will be used to translate
-      the path into the current operating system platform.
+    - If a local storage has been defined, but an operating system is missing,
+      this can be supplied via an environment variable. For example, if there
+      is a local storage named ``Renders`` that is defined on Linux and Windows,
+      you can extend to support mac by creating an environment variable named
+      ``SHOTGUN_PATH_MAC_RENDERS``. The general syntax for this is
+      ``SHOTGUN_PATH_WINDOWS|MAC|LINUX_STORAGENAME``.
 
     - If no root matches, the file path will be returned as is.
 
-    For example, you have published the file ``/projects/some/file.txt`` on Linux
-    and generated a publish with the url ``file:///projects/some/file.txt``. You have
-    either a local storage or environment variable set up with the following paths:
+    .. note::   For example, you have published the file ``/projects/some/file.txt`` on Linux
+                and a Shotgun publish with the url ``file:///projects/some/file.txt`` was generated.
+                The Linux path ``/projects`` equates to ``Q:\projects`` on Windows
+                and hence you want the full path to be translated to
+                ``Q:\\projects\\some\\file.txt``. All of the setups below would handle this:
 
-    - Linux Path: ``/projects``
-    - Windows Path: ``Q:\projects``
-    - Mac Path: ``/projects``
+                - A general environment based override:
 
-    When running on Windows, the ``file://`` url will therefore be translated to
-    ``Q:\\projects\\some\\file.txt``.
+                    - ``SHOTGUN_PATH_LINUX=/projects``
+                    - ``SHOTGUN_PATH_WINDOWS=Q:\projects``
+                    - ``SHOTGUN_PATH_MAC=/projects``
+
+                - A Shotgun local storage "Projects" set up with:
+
+                    - Linux Path: ``/projects``
+                    - Windows Path: ``Q:\projects``
+                    - Mac Path: ``/projects``
+
+                - A Shotgun local storage "Projects" augmented with an environment variable:
+
+                    - Linux Path: ``/projects``
+                    - Windows Path: ``<blank>``
+                    - Mac Path: ``/projects``
+                    - ``SHOTGUN_PATH_WINDOWS_PROJECTS=Q:\projects``
+
+
+    .. note:: If you have a local storage ``Renders`` defined in Shotgun with
```

Comment:
yep, absolutely - this is a special case. the whole process is backwards, in that you start with a `file://` url, then start defining env vars, and the finally solidify your setup by adding local storages in shotgun. so that zero config can grow organically over time. So yes, it's... backwards.